tests/sqlparser_duckdb.rs

```diff
@@ -32,6 +32,38 @@ fn duckdb_and_generic() -> TestedDialects {
     }
 }
 
+#[test]
+fn test_struct() {
+    // basic struct
+    let sql = r#"CREATE TABLE t1 (s STRUCT(v VARCHAR, i INTEGER))"#;
+    let statement = duckdb().verified_stmt(sql);
+    assert_eq!(statement.to_string(), sql);
+
+    // struct array
+    let sql = r#"CREATE TABLE t1 (s STRUCT(v VARCHAR, i INTEGER)[])"#;
+    let statement = duckdb().verified_stmt(sql);
+    assert_eq!(statement.to_string(), sql);
+
+    // nested struct
+    let sql = r#"CREATE TABLE t1 (s STRUCT(v VARCHAR, s STRUCT(a1 INTEGER, a2 VARCHAR))[])"#;
+    let statement = duckdb().verified_stmt(sql);
+    assert_eq!(statement.to_string(), sql);
```

Comment:
```suggestion
    let statement = duckdb().verified_stmt(sql);

    // struct array
    let sql = r#"CREATE TABLE t1 (s STRUCT(v VARCHAR, i INTEGER)[])"#;
    let statement = duckdb().verified_stmt(sql);

    // nested struct
    let sql = r#"CREATE TABLE t1 (s STRUCT(v VARCHAR, s STRUCT(a1 INTEGER, a2 VARCHAR))[])"#;
    let statement = duckdb().verified_stmt(sql);
```
we can remove the explicit string asserts, they're covered by [verified_stmt](https://github.com/sqlparser-rs/sqlparser-rs/blob/main/src/test_utils.rs#L128-L130)